### DISCUSSION TITLE
Discord bot - Python vs JavaScript

### DISCUSSION THREAD

**Discussion by ykdojo** [csdojo-defaang/dojo-bot#3] [General]
Which one's a better choice for this - Python or JavaScript?

Edit: thank you for your input! Let's go with JavaScript or TypeScript.

**Comment by theskollie** [comment]
TypeScript

**Comment by prateek041** [comment]
From my personal experience, Javascript/typescript. let's see what the others have to say.

**Comment by mrrobot-ec** [comment]
I haven't done anything related with discord bots but found this post. https://soyokaze.medium.com/build-a-discord-bot-in-node-or-python-3a4028099657

**Comment by mezotv** [comment]
Been working with both python and javascript in the past and my honest choice for bots would be javascript with discord.js!

**Comment by cybardev** [comment]
As much as I love Python, I'd say go for JavaScript (preferably TypeScript) if you can. Manipulating JSON is much simpler and more natural with JS than with Python. Plus, Python's class-based OOP, as powerful as it is, would make the code more complex than it needs to be.

**Comment by itsanwartarique** [comment]
Python
